### PR TITLE
Merge beta stage arch with architecture changes

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/expenses/components/pending-list/pending-list.component.ts [hunk 1]
```diff
@@ -63,6 +65,25 @@ export class PendingListComponent implements OnInit, OnChanges {
         this.pettyCashReportsResponse$ = this.store.select(p => p.expense.pettycashReport).pipe(takeUntil(this.destroyed$));
         this.getPettycashReportInprocess$ = this.store.select(p => p.expense.getPettycashReportInprocess).pipe(takeUntil(this.destroyed$));
         this.getPettycashReportSuccess$ = this.store.select(p => p.expense.getPettycashReportSuccess).pipe(takeUntil(this.destroyed$));
+
+        observableCombineLatest(this.universalDate$, this.todaySelected$).pipe(takeUntil(this.destroyed$)).subscribe((resp: any[]) => {
+			if (!Array.isArray(resp[0])) {
+				return;
+			}
+			let dateObj = resp[0];
+			this.todaySelected = resp[1];
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/expenses/components/pending-list/pending-list.component.ts:74]: nullity check for resp

File: apps/web-giddh/src/app/expenses/expenses.component.ts [hunk 2]
```diff
@@ -132,28 +148,37 @@ export class ExpensesComponent implements OnInit, OnChanges, OnDestroy {
...
-                    this.pettycashRequest.from = this.unaiversalFrom;
-                    this.pettycashRequest.to = this.unaiversalTo;
+                if (this.universalFrom && this.universalTo) {
+                    this.pettycashRequest.from = this.universalFrom;
+                    this.pettycashRequest.to = this.universalTo;
                     this.pettycashRequest.page = 1;
                     this.selectedDate.dateFrom = this.pettycashRequest.from;
                     this.selectedDate.dateTo = this.pettycashRequest.to;
+
+                    if(this.pendingListComponent) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/expenses/expenses.component.ts:172]: nullity check for pettycashRequest

File: apps/web-giddh/src/app/inventory/components/new-branch-transfer/new.branch.transfer.add.component.ts [hunk 3]
```diff
@@ -545,7 +545,7 @@ export class NewBranchTransferAddComponent implements OnInit, OnChanges, OnDestr
 
     public getWarehouseDetails(type, index): void {
         if (this.branchTransfer[type][index].warehouse.uniqueName !== null) {
-            this._warehouseService.getWarehouseDetails(this.branchTransfer[type][index].uniqueName, this.branchTransfer[type][index].warehouse.uniqueName).subscribe((res) => {
+            this._warehouseService.getWarehouseDetails(this.branchTransfer[type][index].warehouse.uniqueName).subscribe((res) => {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/inventory/components/new-branch-transfer/new.branch.transfer.add.component.ts:548]: nullity check for "type" params

**Review by rohit-walkover — APPROVED** [review]

**Review by shalinee-walkover — APPROVED** [review]
Approve

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**